src/dl.cpp

```diff
@@ -49,12 +51,12 @@ dl_error_t dl_instance_load( dl_ctx_t             dl_ctx,          dl_typeid_t
 {
 	dl_data_header* header = (dl_data_header*)packed_instance;
 
-	if( packed_instance_size < sizeof(dl_data_header) ) return DL_ERROR_MALFORMED_DATA;
-	if( header->id == DL_INSTANCE_ID_SWAPED )           return DL_ERROR_ENDIAN_MISMATCH;
-	if( header->id != DL_INSTANCE_ID )                  return DL_ERROR_MALFORMED_DATA;
-	if( header->version != DL_INSTANCE_VERSION )        return DL_ERROR_VERSION_MISMATCH;
-	if( header->root_instance_type != type_id )         return DL_ERROR_TYPE_MISMATCH;
-	if( header->instance_size > instance_size )         return DL_ERROR_BUFFER_TOO_SMALL;
+	if( packed_instance_size < sizeof(dl_data_header) )                  return DL_ERROR_MALFORMED_DATA;
+	if( header->id == DL_INSTANCE_ID_SWAPED )                            return DL_ERROR_ENDIAN_MISMATCH;
+	if( header->id != DL_INSTANCE_ID )                                   return DL_ERROR_MALFORMED_DATA;
+	if( header->version != DL_INSTANCE_VERSION && header->version != 1 ) return DL_ERROR_VERSION_MISMATCH;
```

Comment:
why do you have to check the != 1 here? DL_INSTANCE_VERSION is 2 now so that will automatically be false right?

Reply:
I reasoned that perhaps someone is already using datalibrary and thus have data saved. To be able to upgrade that data from version 1 to version 2 the code needs to be able to read both during a transition period, so the code accepts version 1 and 2 right now. Feel free to remove that support whenever you want, or refactor that conversion to a separate tool. We don't need it, but I thought it would be kind to others.

Reply:
I'm no longer at Arrowhead unfortunately, but when I was there we rebuild all of the dl-instances frequently as soon as any serializable struct was changed. Which was very frequent. Rebuilding all the dl-instances took minimal amount of time so that was never a problem to do basically after each code-pull. We had almost no pointers so we never had any performance issues with pointer-patching.